### PR TITLE
Ks/p2022 1064

### PR DESCRIPTION
**Pull request by xBonioo**
- CRUD has been edited
- Change IssueDto to BaseIssueDto in create issue
- Adding comments to the issue controller
- Adaptation of methods issue controller to others (it is about removing updateLight, delete, etc.)
- Assign user to issue
- MsSQL and Postgre migrations
- Change when creating issue (nullable add)
- Added if (! Issue.IsActive) conditions so that they no longer show removed ones

### CODE REVIEW

File: src/Patronage.Api/Controllers/IssueController.cs [hunk 1]
```diff
@@ -152,5 +187,32 @@ public async Task<ActionResult> Delete([FromRoute] int issueId)
                 Data = result
             });
         }
+
+        /// <summary>
+        /// Issue assign to user.
```
Comment (Archanial) [inline src/Patronage.Api/Controllers/IssueController.cs:192]:
```suggestion
        /// Assigns issue to user.
```

File: src/Patronage.Api/Controllers/IssueController.cs [hunk 2]
```diff
@@ -152,5 +187,32 @@ public async Task<ActionResult> Delete([FromRoute] int issueId)
...
+        [HttpPut("{issueId}/assign/{userId}")]
+        public async Task<ActionResult> Assign([FromRoute] int issueId, [FromRoute] string userId)
+        {
+            var result = await _mediator.Send(new AssignIssueCommand(issueId, userId));
+            if (!result)
+            {
+                return NotFound(new BaseResponse<bool>
+                {
+                    ResponseCode = StatusCodes.Status404NotFound,
+                    Message = $"There's no issue with Id: {issueId} or no user with Id: {userId}"
```
Comment (Archanial) [inline src/Patronage.Api/Controllers/IssueController.cs:206]:
```suggestion
                    Message = "Issue or user with given Id not found"
```
sounds better imo

File: src/Patronage.Contracts/ModelDtos/Issues/IssueDto.cs [hunk 3]
```diff
@@ -1,25 +1,24 @@
...
-            CreatedOn = issue.CreatedOn;
-            ModifiedOn = issue.ModifiedOn;
             IsActive = issue.IsActive;
+            ModifiedOn = issue.ModifiedOn;
         }
 
+        public IssueDto(BaseIssueDto baseIssue) : base(baseIssue)
+        {
+
+        }
```
Comment (Archanial) [inline src/Patronage.Contracts/ModelDtos/Issues/IssueDto.cs:21]: Why is it empty?

File: src/Patronage.DataAccess/Services/IssueService.cs [hunk 4]
```diff
@@ -160,6 +179,27 @@ public async Task<bool> DeleteAsync(int issueId)
...
+        {
+            var issue = await GetByIdAsync(issueId);
+            if (issue == null)
+            {
+                return false;
+            }
+            if (!issue.IsActive)
+            {
+                return false;
+            }
```
Comment (Archanial) [inline src/Patronage.DataAccess/Services/IssueService.cs:192]:
```suggestion
            if (issue == null || !issue.IsActive)
            {
                return false;
            }
```

File: src/Patronage.Api/Controllers/IssueController.cs (hunk 1 above)
Comment (xBonioo) [reply src/Patronage.Api/Controllers/IssueController.cs]: Done

File: src/Patronage.Api/Controllers/IssueController.cs [hunk 5]
```diff
@@ -1,15 +1,15 @@
 ﻿using MediatR;
 using Microsoft.AspNetCore.Mvc;
-using Patronage.Api.MediatR.Issues.Commands.CreateIssue;
-using Patronage.Api.MediatR.Issues.Commands.DeleteIssue;
-using Patronage.Api.MediatR.Issues.Commands.LightUpdateIssue;
-using Patronage.Api.MediatR.Issues.Commands.UpdateIssue;
-using Patronage.Api.MediatR.Issues.Queries.GetIssues;
-using Patronage.Api.MediatR.Issues.Queries.GetSingleIssue;
+using Patronage.Api.MediatR.Issues.Commands.Create;
```
Comment (mateuszgiedzinski) [inline src/Patronage.Api/Controllers/IssueController.cs:3]: I'm wondering if its necessery to put every command and every query to separate folders(namespaces).  What are pros and cons of thi?.

File: src/Patronage.Api/Validators/Issues/IssueDtoValidator.cs [hunk 6]
```diff
@@ -38,14 +38,11 @@ public IssueDtoValidator(TableContext dbContext)
...
-                {
-                    var isExistProject = dbContext.Issues.Any(p => p.ProjectId == value);
-                    if (!isExistProject)
-                    {
-                        context.AddFailure("ProjectId", "This project id has not been already exist");
-                    }
-                });*/
+
+            RuleFor(x => x.StatusId)
+                .NotNull().WithMessage("Can not be null.")
```
Comment (mateuszgiedzinski) [inline src/Patronage.Api/Validators/Issues/IssueDtoValidator.cs:43]: Not Null has default message better than "Can not be null".

File: src/Patronage.Api/Controllers/IssueController.cs (hunk 5 above)
Comment (xBonioo) [reply src/Patronage.Api/Controllers/IssueController.cs]: Done (both)

**Review by Papuzka — APPROVED** [review]

**Pull request merged**